charm_test.go

```diff
@@ -219,3 +241,98 @@ func cloneDir(c *gc.C, path string) string {
 	c.Assert(err, gc.IsNil)
 	return newPath
 }
+
+// TestCreateMaybeCreateVersionFile verifies if the version file can be created
+// in case of git revision control directory
+func (s *CharmSuite) TestGitMaybeCreateVersionFile(c *gc.C) {
+	// Read the charmDir from the testing folder
+	dummyPath := charmDirPath(c, "dummy")
+
+	testing.PatchExecutableAsEchoArgs(c, s, "git")
+
+	// copy all the contents from 'path' to 'tmp folder dummy-charm'
+	// Using cloneDir
+	tempPath := cloneDir(c, dummyPath)
+
+	// create an empty .git file inside tempDir
+	gitPath := filepath.Join(tempPath, ".git")
+	_, err := os.Create(gitPath)
+	c.Assert(err, gc.IsNil)
+
+	err = charm.MaybeCreateVersionFile(tempPath)
+	c.Assert(err, gc.IsNil)
+
+	testing.AssertEchoArgs(c, "git", expectedArgs.gitBase...)
+
+	versionPath := filepath.Join(tempPath, "version")
+	_, err = os.Stat(versionPath)
```

Comment:
We want to read the contents of the version path file and check that the contents are as expected

Reply:
done